TestChallengerCompleteExhaustiveDisputeGame/RootIncorrect flakes (log after close)

Flaky Test
https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/35246/workflows/6658e076-c3d6-4a0b-8c15-7d05c867e0fc/jobs/1602054/parallel-runs/1
```
panic: Log in goroutine after TestChallengerCompleteExhaustiveDisputeGame/RootIncorrect has completed:             ERROR[12-05|05:38:46.402] error getting latest header              role=Challenger err="Post \"http://127.0.0.1:43741\": dial tcp 127.0.0.1:43741: connect: connection refused"


goroutine 55525 [running]:
testing.(*common).logDepth(0xc000683520, {0xc007e63c80, 0xbc}, 0x3)
	/usr/local/go/src/testing/testing.go:1022 +0x4c5
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:1004
testing.(*common).Logf(0xc000683520, {0x1e79094?, 0x410c85?}, {0xc00ece1500?, 0x1ac03e0?, 0xc007bc6701?})
	/usr/local/go/src/testing/testing.go:1055 +0x54
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).flush(0xc007105b00)
	/root/project/op-service/testlog/testlog.go:164 +0x9c
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).Error(0xc007105b00, {0x20102e9, 0x1b}, {0xc01a0603e0, 0x2, 0x2})
	/root/project/op-service/testlog/testlog.go:130 +0xdb
github.com/ethereum-optimism/optimism/op-service/client.(*PollingClient).pollHeads(0xc01895b900)
	/root/project/op-service/client/polling.go:157 +0x4e7
created by github.com/ethereum-optimism/optimism/op-service/client.NewPollingClient in goroutine 55
	/root/project/op-service/client/polling.go:70 +0x205
panic: Log in goroutine after TestChallengerCompleteExhaustiveDisputeGame/RootCorrect has completed:             ERROR[12-05|05:38:46.541] error getting latest header              role=Challenger err="Post \"http://127.0.0.1:44541\": dial tcp 127.0.0.1:44541: connect: connection refused"


goroutine 55704 [running]:
testing.(*common).logDepth(0xc000c80680, {0xc007e63e00, 0xbc}, 0x3)
	/usr/local/go/src/testing/testing.go:1022 +0x4c5
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:1004
testing.(*common).Logf(0xc000c80680, {0x1e79094?, 0x410c85?}, {0xc00ece1c50?, 0x1ac03e0?, 0xc007bc6a01?})
	/usr/local/go/src/testing/testing.go:1055 +0x54
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).flush(0xc005ae4090)
	/root/project/op-service/testlog/testlog.go:164 +0x9c
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).Error(0xc005ae4090, {0x20102e9, 0x1b}, {0xc01a060660, 0x2, 0x2})
	/root/project/op-service/testlog/testlog.go:130 +0xdb
github.com/ethereum-optimism/optimism/op-service/client.(*PollingClient).pollHeads(0xc011503380)
	/root/project/op-service/client/polling.go:157 +0x4e7
created by github.com/ethereum-optimism/optimism/op-service/client.NewPollingClient in goroutine 41
	/root/project/op-service/client/polling.go:70 +0x205
panic: Log in goroutine after TestChallengerCompleteDisputeGame/DefenderIncorrectAtLastTraceIndex has completed:             ERROR[12-05|05:38:46.406] error getting latest header              role=Defender   err="Post \"http://127.0.0.1:39453\": dial tcp 127.0.0.1:39453: connect: connection refused"


goroutine 51701 [running]:
testing.(*common).logDepth(0xc000557040, {0xc0022a7440, 0xbc}, 0x3)
	/usr/local/go/src/testing/testing.go:1022 +0x4c5
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:1004
testing.(*common).Logf(0xc000557040, {0x1e79094?, 0x410c85?}, {0xc01b8d5680?, 0x1ac03e0?, 0xc00784f901?})
	/usr/local/go/src/testing/testing.go:1055 +0x54
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).flush(0xc015eafc80)
	/root/project/op-service/testlog/testlog.go:164 +0x9c
github.com/ethereum-optimism/optimism/op-service/testlog.(*logger).Error(0xc015eafc80, {0x20102e9, 0x1b}, {0xc00179bb20, 0x2, 0x2})
	/root/project/op-service/testlog/testlog.go:130 +0xdb
github.com/ethereum-optimism/optimism/op-service/client.(*PollingClient).pollHeads(0xc002319680)
	/root/project/op-service/client/polling.go:157 +0x4e7
created by github.com/ethereum-optimism/optimism/op-service/client.NewPollingClient in goroutine 4634
	/root/project/op-service/client/polling.go:70 +0x205
FAIL	github.com/ethereum-optimism/optimism/op-e2e/faultproofs	294.907s
```

--- a/op-challenger/game/service.go
+++ b/op-challenger/game/service.go
@@ -228,6 +228,9 @@ func (s *Service) Stop(ctx context.Context) error {
 		s.txMgr.Close()
 	}
 
+	if s.pollClient != nil {
+		s.pollClient.Close()
+	}
 	if s.l1Client != nil {
 		s.l1Client.Close()
 	}

